src/modules/hip/kernel/bitwise_and.hpp

```diff
@@ -0,0 +1,278 @@
+    dst_f8->f1[1] = (float)((uchar)(src1_f8->f1[1]) & (uchar)(src2_f8->f1[1]));
+    dst_f8->f1[2] = (float)((uchar)(src1_f8->f1[2]) & (uchar)(src2_f8->f1[2]));
+    dst_f8->f1[3] = (float)((uchar)(src1_f8->f1[3]) & (uchar)(src2_f8->f1[3]));
+    dst_f8->f1[4] = (float)((uchar)(src1_f8->f1[4]) & (uchar)(src2_f8->f1[4]));
+    dst_f8->f1[5] = (float)((uchar)(src1_f8->f1[5]) & (uchar)(src2_f8->f1[5]));
+    dst_f8->f1[6] = (float)((uchar)(src1_f8->f1[6]) & (uchar)(src2_f8->f1[6]));
+    dst_f8->f1[7] = (float)((uchar)(src1_f8->f1[7]) & (uchar)(src2_f8->f1[7]));
+}
+
+__device__ void bitwise_and_hip_compute(Rpp8s *srcPtr, d_float8 *src1_f8, d_float8 *src2_f8, d_float8 *dst_f8)
+{
+    dst_f8->f1[0] = (float)((uchar)(src1_f8->f1[0] + (float)128) & (uchar)(src2_f8->f1[0] + (float)128));
+    dst_f8->f1[1] = (float)((uchar)(src1_f8->f1[1] + (float)128) & (uchar)(src2_f8->f1[1] + (float)128));
+    dst_f8->f1[2] = (float)((uchar)(src1_f8->f1[2] + (float)128) & (uchar)(src2_f8->f1[2] + (float)128));
+    dst_f8->f1[3] = (float)((uchar)(src1_f8->f1[3] + (float)128) & (uchar)(src2_f8->f1[3] + (float)128));
+    dst_f8->f1[4] = (float)((uchar)(src1_f8->f1[4] + (float)128) & (uchar)(src2_f8->f1[4] + (float)128));
+    dst_f8->f1[5] = (float)((uchar)(src1_f8->f1[5] + (float)128) & (uchar)(src2_f8->f1[5] + (float)128));
+    dst_f8->f1[6] = (float)((uchar)(src1_f8->f1[6] + (float)128) & (uchar)(src2_f8->f1[6] + (float)128));
+    dst_f8->f1[7] = (float)((uchar)(src1_f8->f1[7] + (float)128) & (uchar)(src2_f8->f1[7] + (float)128));
+    rpp_hip_math_subtract8_const(dst_f8, dst_f8, (float4)128);
+}
+
+__device__ void bitwise_and_hip_compute(half *srcPtr, d_float8 *src1_f8, d_float8 *src2_f8, d_float8 *dst_f8)
+{
+    rpp_hip_math_multiply8_const(src1_f8, src1_f8, (float4)255);
+    rpp_hip_math_multiply8_const(src2_f8, src2_f8, (float4)255);
+    dst_f8->f1[0] = (float)((uchar)(src1_f8->f1[0]) & (uchar)(src2_f8->f1[0]));
+    dst_f8->f1[1] = (float)((uchar)(src1_f8->f1[1]) & (uchar)(src2_f8->f1[1]));
+    dst_f8->f1[2] = (float)((uchar)(src1_f8->f1[2]) & (uchar)(src2_f8->f1[2]));
+    dst_f8->f1[3] = (float)((uchar)(src1_f8->f1[3]) & (uchar)(src2_f8->f1[3]));
+    dst_f8->f1[4] = (float)((uchar)(src1_f8->f1[4]) & (uchar)(src2_f8->f1[4]));
+    dst_f8->f1[5] = (float)((uchar)(src1_f8->f1[5]) & (uchar)(src2_f8->f1[5]));
+    dst_f8->f1[6] = (float)((uchar)(src1_f8->f1[6]) & (uchar)(src2_f8->f1[6]));
+    dst_f8->f1[7] = (float)((uchar)(src1_f8->f1[7]) & (uchar)(src2_f8->f1[7]));
+    rpp_hip_math_multiply8_const(dst_f8, dst_f8, (float4)ONE_OVER_255);
+}
+
+__device__ void bitwise_and_hip_compute(Rpp32f *srcPtr, d_float8 *src1_f8, d_float8 *src2_f8, d_float8 *dst_f8)
+{
+    rpp_hip_math_multiply8_const(src1_f8, src1_f8, (float4)255);
+    rpp_hip_math_multiply8_const(src2_f8, src2_f8, (float4)255);
+    dst_f8->f1[0] = (float)((uchar)(src1_f8->f1[0]) & (uchar)(src2_f8->f1[0]));
+    dst_f8->f1[1] = (float)((uchar)(src1_f8->f1[1]) & (uchar)(src2_f8->f1[1]));
```

Comment:
Please verify with OpenVX, OpenCV and previous RPP if floats are being converted to uchars like here

Reply:
Found this link wherein they convert floats to integers before bitwise operation
https://stackoverflow.com/questions/1723575/how-to-perform-a-bitwise-operation-on-floating-point-numbers

Usually bitwise operations are performed on integers or char datatypes and not directly on float.
I have also checked to see if anyone in our team have used bitwise operations with float value in trainings and didn't find any such case so far.